numbat/src/typechecker.rs

```diff
@@ -297,6 +301,30 @@ pub enum TypeCheckError {
 
     #[error("Base units can not be dimensionless.")]
     NoDimensionlessBaseUnit(Span, String),
+
+    #[error("Unknown struct '{1}")]
+    UnknownStruct(Span, String),
+
+    #[error("Unknown field {2} of struct {3}")]
```

Comment:
I prefer to always add quotes around variable parts of error messages, because they can otherwise lead to confusing sentences with field/struct names that do not immediately stand out:
```suggestion
    #[error("Unknown field '{2}' of struct '{3}'")]
```